pyvrp/cpp/Individual.cpp

```diff
@@ -98,21 +98,23 @@ Individual::Individual(ProblemData const &data, XorShift128 &rng)
     auto const numClients = data.numClients();
     auto const perVehicle = std::max<size_t>(numClients / numVehicles, 1);
     auto const perRoute = perVehicle + (numClients % numVehicles != 0);
+    auto const numRoutes = (numClients + perRoute - 1) / perRoute;
 
-    std::vector<std::vector<Client>> routes(data.numVehicles());
+    std::vector<std::vector<Client>> routes(numRoutes);
     for (size_t idx = 0; idx != numClients; ++idx)
         routes[idx / perRoute].push_back(clients[idx]);
 
-    for (size_t idx = 0; idx != routes.size(); ++idx)
+    for (size_t idx = 0; idx != numRoutes; ++idx)
         routes_[idx] = Route(data, routes[idx]);
+    routes_.resize(numRoutes);
```

Comment:
Yes, if we have 3 vehicles and 4 clients, apparently we create 2 routes with 2 clients, see `test_random_constructor_cycles_over_routes` in `test_Individual.py`, so we need to resize.

Reply:
I think we can reserve sufficient space and `emplace_back` here, like in the other constructor. That should also be a little faster since it saves some constructor calls.

Reply:
I doubt if it will make a difference :p but done